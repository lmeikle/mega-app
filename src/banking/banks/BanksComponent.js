import React from 'react';
import { Link } from 'react-router-dom';
import { string } from 'prop-types';
import { Icon } from 'semantic-ui-react';
import logo from './icon.svg';
import './Banks.css';

/**
 * Stateless component which renders a single Bank component.
 * Clicking on it will show a list of closest ATM's for the selected bank.
 */
const BanksComponent = ({ name, url }) => (
  <Link to={{ pathname: '/banking/atms', state: { name, url } }}>
    <div className="banks-item">
      <img src={logo} className="banks-logo" alt="logo" />
      <div className="banks-name">{name}</div>
      <Icon name="angle right" color="grey" size="large" className="banks-arrow" />
    </div>
  </Link>
);

BanksComponent.propTypes = {
  name: string.isRequired,
  url: string.isRequired
};

BanksComponent.defaultProps = {};

export default BanksComponent;
