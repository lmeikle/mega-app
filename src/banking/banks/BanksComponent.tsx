import React, { SFC } from 'react';
import { Link } from 'react-router-dom';
import { Icon } from 'semantic-ui-react';
import { BankProps } from './BanksTypes';
import logo from './icon.svg';
import './Banks.css';

/**
 * Stateless component which renders a single Bank component.
 * Clicking on it will show a list of closest ATM's for the selected bank.
 */
const BanksComponent: SFC<BankProps> = ({ name, url }) => (
  <Link to={{ pathname: '/banking/atms', state: { name, url } }}>
    <div className="banks-item">
      <img src={logo} className="banks-logo" alt="logo" />
      <div className="banks-name">{name}</div>
      <Icon name="angle right" color="grey" size="large" className="banks-arrow" />
    </div>
  </Link>
);

export default BanksComponent;
