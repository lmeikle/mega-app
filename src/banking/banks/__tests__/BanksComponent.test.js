import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import BanksComponent from '../BanksComponent';

let testProps = {
  name: 'Barclays Bank',
  url: 'https://www.example.com'
};

const linkToProps = { pathname: '/banking/atms', state: testProps };

describe('BanksComponent', () => {
  test('renders correctly', () => {
    const banksComponent = shallow(<BanksComponent {...testProps} />);
    expect(banksComponent).toMatchSnapshot();
  });

  test('includes link to atms page', () => {
    const wrapper = mount(
      <MemoryRouter>
        <BanksComponent {...testProps} />
      </MemoryRouter>
    );

    expect(wrapper.find('Link').props().to).toEqual(linkToProps);
  });
});
