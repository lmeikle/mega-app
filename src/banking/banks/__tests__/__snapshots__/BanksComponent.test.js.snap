// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BanksComponent renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BanksComponent
    name="Barclays Bank"
    url="https://www.example.com"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="banks-item"
      >
        <img
          alt="logo"
          className="banks-logo"
          src="icon.svg"
        />
        <div
          className="banks-name"
        >
          Barclays Bank
        </div>
        <Icon
          as="i"
          className="banks-arrow"
          color="grey"
          name="angle right"
          size="large"
        />
      </div>,
      "replace": false,
      "to": Object {
        "pathname": "/banking/atms",
        "state": Object {
          "name": "Barclays Bank",
          "url": "https://www.example.com",
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="banks-logo"
            src="icon.svg"
          />,
          <div
            className="banks-name"
          >
            Barclays Bank
          </div>,
          <Icon
            as="i"
            className="banks-arrow"
            color="grey"
            name="angle right"
            size="large"
          />,
        ],
        "className": "banks-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "banks-logo",
            "src": "icon.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Barclays Bank",
            "className": "banks-name",
          },
          "ref": null,
          "rendered": "Barclays Bank",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "className": "banks-arrow",
            "color": "grey",
            "name": "angle right",
            "size": "large",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="banks-item"
        >
          <img
            alt="logo"
            className="banks-logo"
            src="icon.svg"
          />
          <div
            className="banks-name"
          >
            Barclays Bank
          </div>
          <Icon
            as="i"
            className="banks-arrow"
            color="grey"
            name="angle right"
            size="large"
          />
        </div>,
        "replace": false,
        "to": Object {
          "pathname": "/banking/atms",
          "state": Object {
            "name": "Barclays Bank",
            "url": "https://www.example.com",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="banks-logo"
              src="icon.svg"
            />,
            <div
              className="banks-name"
            >
              Barclays Bank
            </div>,
            <Icon
              as="i"
              className="banks-arrow"
              color="grey"
              name="angle right"
              size="large"
            />,
          ],
          "className": "banks-item",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "banks-logo",
              "src": "icon.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Barclays Bank",
              "className": "banks-name",
            },
            "ref": null,
            "rendered": "Barclays Bank",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "className": "banks-arrow",
              "color": "grey",
              "name": "angle right",
              "size": "large",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
